ggshield/git_shell.py

```diff
@@ -12,11 +12,21 @@
 
 @lru_cache(None)
 def get_git_path(cwd: str) -> str:
-    git_path = str(which("git"))
+    git_path = which("git")
 
-    if cwd in git_path and cwd not in os.environ.get("PATH", "").split(os.pathsep):
+    if git_path is None:
         raise Exception("unable to find git executable in PATH/PATHEXT")
 
+    # lower()ing these would provide additional coverage on case-
+    # insensitive filesystems but detection is problematic
+    git_path = os.path.abspath(git_path)
+    cwd = os.path.abspath(cwd)
+    path_env = list(map(os.path.abspath, os.environ.get("PATH", "").split(os.pathsep)))
```

Comment:
It does seem to be a linter issue (https://github.com/python/mypy/issues/9864) 

I see two solutions
```py
path_env = list(map(os.path.abspath, os.environ.get("PATH", "").split(os.pathsep)))  # type: ignore # noqa
```
mypy ignore it and then ignore the too long line

or

```py
path_env = [
        os.path.abspath(path) for path in os.environ.get("PATH", "").split(os.pathsep)
    ]
```

use a list comprehension, I'm not too worried about the performance impact of either since this is not bottleneck sensitive code.

Up to you how you want to change it :slightly_smiling_face:

Reply:
Updated and pushed - looks like mypy is happy now.